qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,550 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Base class of curve analysis.
+"""
+
+import warnings
+
+from abc import ABC, abstractmethod
+from typing import List, Dict, Union
+
+import numpy as np
+from uncertainties import unumpy as unp
+
+from qiskit_experiments.framework import BaseAnalysis, AnalysisResultData, Options, ExperimentData
+from qiskit_experiments.data_processing import DataProcessor
+from qiskit_experiments.data_processing.processor_library import get_processor
+from qiskit_experiments.data_processing.exceptions import DataProcessorError
+from qiskit_experiments.exceptions import AnalysisError
+
+from .curve_data import CurveData, SeriesDef, FitData, ParameterRepr, FitOptions
+from .data_processing import multi_mean_xy_data, data_sort
+from .curve_fit import multi_curve_fit
+from .visualization import MplCurveDrawer, BaseCurveDrawer
+
+PARAMS_ENTRY_PREFIX = "@Parameters_"
+DATA_ENTRY_PREFIX = "@Data_"
+
+
+class BaseCurveAnalysis(BaseAnalysis, ABC):
+    """Abstract superclass of curve analysis base classes.
+
+    Note that this class doesn't define :meth:`_run_analysis` method,
+    and no actual fitting protocol is implemented in this base class.
+    However, this class defines several common methods that can be reused.
+    A curve analysis subclass can construct proper fitting protocol
+    by combining following methods, i.e. subroutines.
+    See :ref:`curve_analysis_workflow` for how these subroutines are called.
+
+    .. rubric:: _generate_fit_guesses
+
+    An abstract method to create initial guesses for the fit parameters.
```

Comment:
This method isn't abstract in this class
```suggestion
    A method to create initial guesses for the fit parameters.
```